### ISSUE TITLE
False positive for rule `prefer_self_in_static_references`

### ISSUE THREAD

**Issue by pm-dev** [realm/SwiftLint#5118]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following is a triggering example with this rule enabled, yet it results in a compiler error: 
`error: covariant ‘Self’ type cannot be referenced from a stored property initializer`

```
class A {
        static let start: Int = 0

        private(set) var current: Int = A.start {
             didSet {
                 self.foo()
             }
        }
}
```
This will incorrectly convert `private(set) var current: Int = A.start {` to `private(set) var current: Int = Self.start {`

However this rule works correctly and doesn't trigger without the `didSet` block:
```
class A {
        static let start: Int = 0
        private(set) var current: Int = A.start
}
```

### Environment

* SwiftLint version: 0.52.4
* Installation method used: Homebrew
* Which Xcode version are you using: Xcode 14.3
